community/kernel/src/main/java/org/neo4j/kernel/impl/factory/EditionModule.java

```diff
@@ -96,19 +93,37 @@ protected void publishEditionInfo( UsageData sysInfo, DatabaseInfo databaseInfo,
         config.augment( singletonMap( Configuration.editionName.name(), databaseInfo.edition.toString() ) );
     }
 
-    protected AuthManager createAuthManager(Config config, LifeSupport life, LogProvider logProvider)
+    public static AuthManager createAuthManager( Config config, LogService logging )
     {
-
         boolean authEnabled = config.get( GraphDatabaseSettings.auth_enabled );
-        if ( authEnabled)
+        if ( !authEnabled )
         {
-            FileUserRepository users = life.add( new FileUserRepository( config.get( GraphDatabaseSettings.auth_store ).toPath(), logProvider ) );
-            return life.add(new BasicAuthManager( users, systemUTC(), true ));
+            return AuthManager.NO_AUTH;
         }
-        else
+
+        String key = config.get( GraphDatabaseSettings.auth_manager );
+        for ( AuthManager.Factory candidate : Service.load( AuthManager.Factory.class ) )
         {
-            return AuthManager.NO_AUTH;
+            String candidateId = candidate.getKeys().iterator().next();
+            if ( candidateId.equals( key ) )
+            {
+                return candidate.newInstance( config, logging.getUserLogProvider() );
+            }
+            else if ( key.equals( "" ) )
+            {
+                logging.getInternalLog( CommunityFacadeFactory.class )
+                        .info( "No auth manager implementation specified, defaulting to '" + candidateId + "'" );
+                return candidate.newInstance( config, logging.getUserLogProvider() );
```

Comment:
Like described above, as with the lock manager, the benefit of loading the first available service is that it is easy to configure the different builds just by adding the right maven dependency. For a production system maybe you want this to be locked down, but it could be done by setting the configuration value.